pkg/ibmvpc/sg_analysis.go

```diff
@@ -111,45 +132,82 @@ func (sga *SGAnalyzer) getRemoteCidr(remote vpc1.SecurityGroupRuleRemoteIntf) (*
 	return target, cidrRes, nil
 }
 
+func (sga *SGAnalyzer) getLocalCidr(local vpc1.SecurityGroupRuleLocalIntf) (*ipblock.IPBlock, string, error) {
+	var localIP *ipblock.IPBlock
+	var cidr, address string
+	var cidrRes string
+	var err error
+	if localObj, ok := local.(*vpc1.SecurityGroupRuleLocal); ok {
+		switch {
+		case localObj.CIDRBlock != nil:
+			cidr = *localObj.CIDRBlock
+		case localObj.Address != nil:
+			address = *localObj.Address
+		default:
+			return nil, "", fmt.Errorf("sg error: getLocalCidr - SecurityGroupRuleLocal is empty: %+v", localObj)
+		}
+		localIP, cidrRes, err = sga.getIPBlockResult(cidr, address, "")
+		if err != nil {
+			return nil, "", err
+		}
+
+		if localIP == nil || cidrRes == "" {
+			return localIP, cidrRes, fmt.Errorf("sg error: getLocalCidr returns empty result. localObj: %+v", localObj)
+		}
+	}
+	if localIP == nil || cidrRes == "" {
+		return localIP, cidrRes, fmt.Errorf("sg error: getLocalCidr returns empty result. could not convert localObj to expected type ")
+	}
+	return localIP, cidrRes, nil
+}
+
 func (sga *SGAnalyzer) getProtocolAllRule(ruleObj *vpc1.SecurityGroupRuleSecurityGroupRuleProtocolAll) (
 	ruleStr string, ruleRes *SGRule, isIngress bool, err error) {
 	ruleRes = &SGRule{}
 	direction := *ruleObj.Direction
 	isIngress = isIngressRule(ruleObj.Direction)
 	protocol := *ruleObj.Protocol
-	remote := ruleObj.Remote
-	cidr := ""
-	var target *ipblock.IPBlock
-	target, cidr, err = sga.getRemoteCidr(remote)
-	if err == nil {
-		if target == nil {
-			return "", nil, false, fmt.Errorf("getSGRule error: empty target in rule %+v", ruleObj)
-		}
-		ruleStr := fmt.Sprintf("direction: %s, protocol: %s, cidr: %s\n", direction, protocol, cidr)
-		ruleRes.target = target
-		ruleRes.connections = getAllConnSet()
-		return ruleStr, ruleRes, isIngress, nil
+	remoteCidr, localCidr := "", ""
+	var target, local *ipblock.IPBlock
+	target, remoteCidr, err = sga.getRemoteCidr(ruleObj.Remote)
+	if err != nil {
+		return "", nil, false, err
+	}
+	local, localCidr, err = sga.getLocalCidr(ruleObj.Local)
+	if err != nil {
+		return "", nil, false, err
+	}
+	if target == nil {
+		return "", nil, false, fmt.Errorf("getSGRule error: empty target in rule %+v", ruleObj)
+	}
+	if local == nil {
+		return "", nil, false, fmt.Errorf("getSGRule error: empty local in rule %+v", ruleObj)
```

Comment:
why is this checked only in this function (`getProtocolAllRule`) and not in the similar functions `getProtocolTcpudpRule`, and `getProtocolIcmpRule` ?
also, can this common code be handled in a separate function?

Reply:
fixed, no need for this check, it is checked inside `getLocalCidr` and `getRemoteCidr`